mono/mini/interp/interp.c

```diff
@@ -6407,10 +6415,11 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 
 		MINT_IN_CASE(MINT_INTRINS_ENUM_HASFLAG) {
 			MonoClass *klass = (MonoClass*)imethod->data_items[* (guint16 *)(ip + 1)];
-			guint64 a_val = 0, b_val = 0;
-
-			stackval_to_data (m_class_get_byval_arg (klass), &sp [-2], &a_val, FALSE);
-			stackval_to_data (m_class_get_byval_arg (klass), &sp [-1], &b_val, FALSE);
+			const guint64 a_val = stackval_to_data_enum (m_class_get_byval_arg (klass), &sp [-2]);
+			// Some compilers will use stack to preserve klass across the two calls
+			// to stackval_to_data_enum. Refetch to try to avoid that.
+			klass = (MonoClass*)imethod->data_items[* (guint16 *)(ip + 1)];
+			const guint64 b_val = stackval_to_data_enum (m_class_get_byval_arg (klass), &sp [-1]);
```

Comment:
This might be worth "full outlining" instead of the chatty multiple function calls.

Reply:
I think so too

Reply:
https://github.com/mono/mono/pull/16228